dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -446,10 +447,12 @@ def sign_in_user(user)
   end
 
   def allows_silent_takeover(oauth_user, auth_hash)
-    allow_takeover = auth_hash.provider.present?
-    allow_takeover &= AuthenticationOption::SILENT_TAKEOVER_CREDENTIAL_TYPES.include?(auth_hash.provider.to_s)
+    return false unless auth_hash.provider.present?
+    return false unless AuthenticationOption::SILENT_TAKEOVER_CREDENTIAL_TYPES.include?(auth_hash.provider.to_s)
+    return false if oauth_user.persisted?
+
     lookup_user = User.find_by_email_or_hashed_email(oauth_user.email)
```

Comment:
i think we should add a lookup for users/authentication_options with the same uid/provider here as well

Reply:
and on [line 363](https://github.com/code-dot-org/code-dot-org/blob/4e61395f8b758cc1efb6188d393d156e7690a7cf/dashboard/app/controllers/omniauth_callbacks_controller.rb#L363)

Reply:
I'm actually rethinking that; that's what we settled on yesterday, but I'm digging into it and I'm starting to think that takeovers should _only_ be for the situation where a user already exists with the given email, and that we should resolve the situation where a user already exists with the matching oauth information by doing a better job of preventing the second account from being created in the first place.

Reply:
ah, okay, i think that sounds totally reasonable